ui/src/utils/api-examples/index.js

```diff
@@ -0,0 +1,205 @@
+import generateShellExample from './shell';
+import generatePayloadExample from './payload-generator';
+
+// Default version - should match the current Taskcluster version
+const DEFAULT_VERSION = '93';
+// Import references for schema lookup
+let references = [];
+
+try {
+  // Try to import references - will be available after yarn generate
+  // eslint-disable-next-line global-require
+  references = require('../../../../generated/references.json');
+} catch (e) {
+  // References not available - examples will use placeholder payloads
+}
+
+/**
+ * Get schema content by ID
+ *
+ * @param {string} schemaId - Schema ID
+ * @returns {object|null} Schema content or null if not found
+ */
+function getSchemaContent(schemaId) {
+  if (!references || references.length === 0) {
+    return null;
+  }
+
+  const schema = references.find(({ content }) => content.$id === schemaId);
+
+  return schema ? schema.content : null;
+}
+
+/**
+ * Generate a single code example for a specific language
+ *
+ * @param {string} language - Language key (e.g., 'go', 'python', 'curl')
+ * @param {string} serviceName - Service name (e.g., 'queue', 'auth')
+ * @param {string} apiVersion - API version (e.g., 'v1')
+ * @param {object} entry - API entry metadata from references.json
+ * @param {string} version - Taskcluster version (e.g., '93')
+ * @returns {string|null} Code example or null if not found
+ */
+function generateSingleExample(
+  language,
+  serviceName,
+  apiVersion,
+  entry,
+  version = DEFAULT_VERSION
+) {
+  // Generate payload example if entry has input schema
+  let payloadExample = null;
+
+  if (entry.input) {
+    const schemaId = `/schemas/${serviceName}/${entry.input}`;
+    const schema = getSchemaContent(schemaId);
+
+    if (schema) {
+      // Pass all references for $ref resolution
+      payloadExample = generatePayloadExample(schema, references);
+    }
+  }
+
+  const generators = {
+    curl: () =>
+      generateCurlExample(serviceName, apiVersion, entry, payloadExample),
+    go: () =>
+      generateGoExample(
+        serviceName,
+        apiVersion,
+        entry,
+        version,
+        payloadExample
+      ),
+    python: () =>
+      generatePythonExample(
+        serviceName,
+        apiVersion,
+        entry,
+        false,
+        payloadExample
+      ),
+    pythonAsync: () =>
+      generatePythonExample(
+        serviceName,
+        apiVersion,
+        entry,
+        true,
+        payloadExample
+      ),
+    node: () =>
+      generateNodeExample(serviceName, apiVersion, entry, payloadExample),
+    web: () =>
+      generateWebExample(serviceName, apiVersion, entry, payloadExample),
+    rust: () =>
+      generateRustExample(serviceName, apiVersion, entry, payloadExample),
+    shell: () =>
+      generateShellExample(serviceName, apiVersion, entry, payloadExample),
+  };
+  const generator = generators[language];
+
+  if (!generator) {
+    // Unknown language requested
```

Comment:
I would probably prefer an assert here and early failure or even exception, if someone is using this function incorrectly to avoid silencing errors